airflow-core/src/airflow/dag_processing/collection.py

```diff
@@ -196,7 +196,7 @@ def _serialize_dag_capturing_errors(
             min_update_interval=settings.MIN_SERIALIZED_DAG_UPDATE_INTERVAL,
             session=session,
         )
-        if dag_was_updated:
+        if dag_was_updated and "FabAuthManager" in conf.get("core", "auth_manager"):
```

Comment:
This means we always run `update_source_code` if we don't have FAB. We should refactor this either like this:

```suggestion
        if dag_was_updated:
            if "FabAuthManager" in conf.get("core", "auth_manager"):
```

Reply:
I had to revert the commit. I think it should be with the and since only fab runs sync perms. WDYT?